### PR TITLE
Adding insecure option to disable client certificate verification

### LINKED ISSUE #2001: Need to disable certificate checking within Docker container
I am using the docker command to run mwoffliner on my local medaiwiki server  and i get the UNABLE_TO_GET_ISSUER_CERT_LOCALLY  .

This is the comamnd

$ sudo docker run --volume=$(pwd)/out:/out -ti ghcr.                                               io/openzim/mwoffliner mwoffliner --mwUrl=http://<local ip>  --adminEmail=foo@bar.net --mwWikiPath=/vascular_wiki/ --mwApiPath=vascular_wiki/api.php 

[error] [2024-03-02T12:49:43.043Z] FATAL - Failed to get MediaWiki Metadata
[error] [2024-03-02T12:49:43.046Z] Failed to run mwoffliner after [40s]: {
        "message": "unable to get local issuer certificate",
        "name": "Error",
        "stack": "Error: unable to get local issuer certificate\n    at AxiosError.from (file:///tmp/mwoffliner/node_modules/axios/lib/core/AxiosError.js:89:14)\n    at RedirectableRequest.handleRequestError (file:///tmp/mwoffliner/node_modules/axios/lib/adapters/http.js:577:25)\n    at RedirectableRequest.emit (node:events:539:35)\n    at RedirectableRequest.emit (node:domain:475:12)\n    at eventHandlers.<computed> (/tmp/mwoffliner/node_modules/follow-redirects/index.js:14:24)\n    at ClientRequest.emit (node:events:527:28)\n    at ClientRequest.emit (node:domain:475:12)\n    at TLSSocket.socketErrorListener (node:_http_client:462:9)\n    at TLSSocket.emit (node:events:527:28)\n    at TLSSocket.emit (node:domain:475:12)",

        },
        "code": "UNABLE_TO_GET_ISSUER_CERT_LOCALLY",
        "status": null
}
[error] [2024-03-02T12:49:43.049Z]

**********

unable to get local issuer certificate

### PR DESCRIPTION
**Pull request by jairajmahadev**
Added ```--insecure``` option. When provided, disables SSL certificate verification in axios. Fixes #2001

### CODE REVIEW

**Comment by jairajmahadev** [comment]
How do I fix this issue thrown by codefactor?

**Comment by kelson42** [comment]
@jairajmahadev Thank you for your PR, this error from CodeFactor is not your fault and is not easy to fix, please just ignore it. Put the PR in ready to review when you will be so far.

**Comment by jairajmahadev** [comment]
@kelson42 PR is ready for review. Also, i've added a ``` console.log ``` to test and see the value of the ```insecure``` variable. I had added a ```logger.warn``` log earlier, but the warning was not shown on the console. Need some help with that.

File: src/Downloader.ts [hunk 1]
```diff
@@ -106,7 +107,9 @@ class Downloader {
   private articleUrlDirector: URLDirector
   private mainPageUrlDirector: URLDirector
```
Comment (kelson42) [inline src/Downloader.ts:109]: Why an empty line here?

File: src/Downloader.ts [hunk 2]
```diff
@@ -130,7 +136,7 @@ class Downloader {
     this.arrayBufferRequestOptions = {
       // HTTP agent pools with 'keepAlive' to reuse TCP connections, so it's faster
       httpAgent: new http.Agent({ keepAlive: true }),
-      httpsAgent: new https.Agent({ keepAlive: true }),
+      httpsAgent: new https.Agent({ keepAlive: true, rejectUnauthorized: !this.insecure }), // rejectUnauthorized: false disables SSL verification
```
Comment (kelson42) [inline src/Downloader.ts:139]: You mean TLS (not SSL)

File: src/parameterList.ts [hunk 3]
```diff
@@ -40,6 +40,7 @@ export const parameterDescriptions = {
   optimisationCacheUrl: 'Object Storage URL (including credentials and bucket name) to cache optimised media files',
   forceRender:
     'Force the usage of a specific API end-point/render, automatically chosen otherwise. Accepted values: [ VisualEditor, WikimediaDesktop. WikimediaMobile ]. More details at https://github.com/openzim/mwoffliner/wiki/API-end-points',
+  insecure: 'Allow insecure server connections when using HTTPS',
```
Comment (kelson42) [inline src/parameterList.ts:43]: Better: `Skip HTTPS server authenticity verification step`

File: src/sanitize-argument.ts [hunk 4]
```diff
@@ -35,8 +35,12 @@ export async function sanitize_all(argv: any) {
     mwActionApiPath,
     mwRestApiPath,
     mwModulePath,
+    insecure,
   } = argv
 
+  // set insecure to false by default
```
Comment (kelson42) [inline src/sanitize-argument.ts:41]: This is quite clear, you don't need to put a comment here

Comment (kelson42) [reply src/sanitize-argument.ts:41]: There is nothing to sanitize here. Unfortunately there is not a real place defined looks like for such an initialisation, but I would say either in `src/sanitize-argument.ts` or in Downloader constructor (like `--webp`)

**Review by kelson42 — CHANGES_REQUESTED** [review]
Almost good, only a few changes.

File: src/Downloader.ts (hunk 2 above)
Comment (jairajmahadev) [reply src/Downloader.ts:139]: yes, it disables the SSL certificate verification step in the TLS handshake. So should I just change it to say ```disables TLS```?

File: src/parameterList.ts (hunk 3 above)
Comment (jairajmahadev) [reply src/parameterList.ts:43]: Okay

File: src/sanitize-argument.ts (hunk 4 above)
Comment (jairajmahadev) [reply src/sanitize-argument.ts]: directly in Downloader constructor works

File: src/Downloader.ts [hunk 5]
```diff
@@ -116,6 +118,9 @@ class Downloader {
     this.webp = webp
     this.s3 = s3
     this.apiUrlDirector = new ApiURLDirector(MediaWiki.actionApiUrl.href)
+    this.insecure = insecure
+
+    console.log('Insecure: ', this.insecure) // printing for testing purpose
```
Comment (kelson42) [inline src/Downloader.ts:123]: Please remove that line

**Review by kelson42 — CHANGES_REQUESTED** [review]
Last small thing and we can merge

File: src/Downloader.ts (hunk 5 above)
Comment (jairajmahadev) [reply src/Downloader.ts]: done

**Review by kelson42 — APPROVED** [review]
LGTM

File: src/Downloader.ts (hunk 2 above)
Comment (kelson42) [reply src/Downloader.ts]: Yes

**Pull request merged**